src/components/Map/Map.js

```diff
@@ -29,7 +31,26 @@ const Map = () => {
     setMapView(view);
   };
 
+  const handleMapExtentChanged = async (stationary) => {
+    // If stationary is false, the map is still being moved, abort
+    if (!stationary) return;
+
+    // Since map is now stationary, fetch trails
+    const trails = await fetchTrailsInExtent();
+
+    // If trails is undefined, abort
+    if (!trails) {
+      // TODO: Fix issue where this happens on initial load
+      console.log('no trails!', trails);
+      return;
+    }
+
+    setFeatureList(trails);
+  };
+
   useWatch(view, 'ready', handleMapReady);
+  useWatch(view, 'stationary', debounce(handleMapExtentChanged, 500));
```

Comment:
@odoe this fires once as the app initializes, but returns `undefined` (i think because `app.view` in map.js doesn't exist yet). do you have any ideas for getting the features from the map when it first loads?

Alternatively... maybe we shouldn't show trails on initial load anyway (unless one is selected). Maybe we just turn that layer off in the webmap then turn it on when appropriate in the app?

Reply:
For the starting extent? Let me rework that function for that

Reply:
You have a view there right? I can take the `view` that you have and work from it there

Reply:
that function can now take a view, so you can pass `fetchTrailsInExtent(view)` and it will use the given `view` or check `app.view` if it exists there.

Reply:
Awesome, seems to be working now. Thanks!